source/common/http/filter/squash_filter.cc

```diff
@@ -0,0 +1,340 @@
+      getAttachmentFromStruct(*curvalue.mutable_struct_value());
+    } else {
+      getAttachmentFromValue(curvalue);
+    }
+  }
+}
+
+void SquashFilterConfig::getAttachmentFromValue(ProtobufWkt::Value& curvalue) {
+  switch (curvalue.kind_case()) {
+  case ProtobufWkt::Value::kStructValue: {
+    getAttachmentFromStruct(*curvalue.mutable_struct_value());
+    break;
+  }
+  case ProtobufWkt::Value::kListValue: {
+    ProtobufWkt::ListValue& values = *curvalue.mutable_list_value();
+    for (int i = 0; i < values.values_size(); i++) {
+      getAttachmentFromValue(*values.mutable_values(i));
+    }
+    break;
+  }
+  case ProtobufWkt::Value::kStringValue: {
+    curvalue.set_string_value(replaceEnv(curvalue.string_value()));
+    break;
+  }
+  case ProtobufWkt::Value::KIND_NOT_SET:
+  case ProtobufWkt::Value::kNullValue:
+  case ProtobufWkt::Value::kBoolValue:
+  case ProtobufWkt::Value::kNumberValue: {
+    // nothing here... we only need to transform strings
+  }
+  }
+}
+
+/*
+ This function interpolates environment variables in a string template.
+ To interpolate an environment variable named ENV, add '{{ ENV }}' (without the
+  quotes, with the spaces) to the template string.
+
+  See https://github.com/envoyproxy/data-plane-api/blob/master/api/filter/http/squash.proto#L36 for
+  the motivation on why this is needed.
+*/
+std::string SquashFilterConfig::replaceEnv(const std::string& attachment_template) {
+  std::string s;
+
+  auto end_last_match = attachment_template.begin();
+
+  auto replaceEnvVarInTemplateCallback =
+      [&s, &attachment_template,
+       &end_last_match](const std::match_results<std::string::const_iterator>& match) {
+        auto start_match = attachment_template.begin() + match.position(0);
+
+        s.append(end_last_match, start_match);
+
+        std::string envar_name = match[1].str();
+        const char* envar_value = std::getenv(envar_name.c_str());
+        if (envar_value == nullptr) {
+          ENVOY_LOG(warn, "Squash: no environment variable named {}.", envar_name);
+        } else {
+          s.append(envar_value);
+        }
+        end_last_match = start_match + match.length(0);
+      };
+
+  std::sregex_iterator begin(attachment_template.begin(), attachment_template.end(), ENV_REGEX),
+      end;
+  std::for_each(begin, end, replaceEnvVarInTemplateCallback);
+  s.append(end_last_match, attachment_template.end());
+
+  return s;
+}
+
+SquashFilter::SquashFilter(SquashFilterConfigSharedPtr config, Upstream::ClusterManager& cm)
+    : config_(config), state_(State::Initial), debugAttachmentPath_(), delay_timer_(nullptr),
+      attachment_timeout_timer_(nullptr), in_flight_request_(nullptr), cm_(cm),
+      decoder_callbacks_(nullptr) {}
+
+SquashFilter::~SquashFilter() {}
+
+void SquashFilter::onDestroy() { cleanup(); }
+
+FilterHeadersStatus SquashFilter::decodeHeaders(HeaderMap& headers, bool) {
+
+  // check for squash header
```

Comment:
We're not *always* consistent about this, but would you mind running through these and capitalizing?

https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar